src/libs/actions/Policy.js

```diff
@@ -236,22 +251,37 @@ function createAndGetPolicyList() {
  * @param {String} policyID
  */
 function loadFullPolicy(policyID) {
-    API.GetFullPolicy(policyID)
-        .then((data) => {
-            if (data.jsonCode !== 200) {
-                return;
-            }
-
-            const policy = lodashGet(data, 'policyList[0]', {});
-            if (!policy.id) {
-                return;
-            }
-
-            Onyx.set(`${ONYXKEYS.COLLECTION.POLICY}${policy.id}`, {
-                ...allPolicies[`${ONYXKEYS.COLLECTION.POLICY}${policy.id}`],
-                ...getSimplifiedPolicyObject(policy, true),
+    const key = `${ONYXKEYS.COLLECTION.POLICY}${policyID}`;
+    const getFullPolicy = () => {
+        API.GetFullPolicy(policyID)
+            .then((data) => {
+                if (data.jsonCode !== 200) {
+                    return;
+                }
+
+                const policy = lodashGet(data, 'policyList[0]', {});
+                if (!policy.id) {
+                    return;
+                }
+
+                const currentPolicy = _.clone(allPolicies[key]);
+                const simplifiedPolicyObject = getSimplifiedPolicyObject(policy, true);
+
+                const updatedPolicy = {
+                    ...currentPolicy,
+                    ...simplifiedPolicyObject,
+                    pendingInvitations: [],
+                };
+                Onyx.set(key, updatedPolicy);
             });
-        });
+    };
+
+    if (_.isEmpty(lodashGet(allPolicies[key], 'pendingInvitations', []))) {
+        getFullPolicy();
+    } else {
+        // If there is pendingInvitations, wait until invite user API call finish
+        setTimeout(() => getFullPolicy(), 3000);
+    }
```

Comment:
Why these changes are needed? But this is not the correct place to do this. What is 3000?

Reply:
If there is `pendingInvitations` then wait the API call to finish. Hence `3000`

Reply:
Why do you want to wait for 3 secs? How did you determine this duration? This logic does not explain what is happening here. The API call is already made. Why are you recursive calling the same API again?

```
if (_.isEmpty(lodashGet(allPolicies[key], 'pendingInvitations', []))) {
        getFullPolicy();
    }
```
What if there are no pending invitations `pendingInvitations: []`. This will create a infinite loop.

Reply:
It is not recursive and no infinite loop here. `getFullPolicy` and `loadFullPolicy` is different.

Reply:
Oh Sorry, This is not recursive. It looked like it was called inside `getFullPolicy` inner function.